doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1307,6 +1309,57 @@ Displaying information about scopes
       It also displays the delimiting key if any and the class to which the
       scope is bound, if any.
 
+Impact of scopes on printing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When several notations are available for printing the same expression,
+Coq will use the following rules for printing priorities:
+
+- If two notations are available in different scopes which are open,
+  the notation in the more recently opened scope takes precedence.
+
+- If two notations are available in the same scope, the more recently
+  defined (or imported) notation takes precedence.
+
+- Abbreviations and lonely notations, both of which have no scope,
+  take precedence over a notation in an open scope if and only if the
+  abbreviation or lonely notation was defined (or imported) more
+  recently than when the corresponding scope was open. They take
+  precedences over any notation not in open scope, whether this scope
+  has a delimiter or not.
+
+- A scope is active for printing a term either because it was opened
+  with :cmd:`Open Scope`, or the term is the immediate argument of a
+  constant which temporarily opens a scope for this argument (see
+  :ref:`Arguments <ArgumentScopes>`) in which case this temporary
+  scope is the most recent open one.
+
+- In case no abbreviation, nor lonely notation, nor notation in an
+  explicitly open scope, nor notation in a temporarily open scope of
+  arguments, has been found, notations in those closed scopes which
+  have a delimiter are considered, giving priority to the most
+  recently defined (or imported) ones. The corresponding delimiter is
+  inserted, making the corresponding scope the most recent explicitly
+  open scope for all subterms of the current term. As an exception to
+  the insertion of the corresponding delimiter, when an expression is
+  statically known to be in position of type and the notation is from
+  scope ``type_scope``, and the latter is closed, the delimiter is not
+  inserted. This is because expressions statically known to be in
+  position of type are by default interpreted with `type_scope`
+  temporarily activated.
+
+- As a special rule of the previous item, in case of applied global
+  references, notations in a non-opened scope with delimiter
+  specifically defined for this applied global reference are
+  prioritary over notations in a non-opened scope with delimiter for
+  generic applications (such as e.g. ``Notation "f ( x )" := (f x) (at
```

Comment:
This needs a more complete example.  I think I understand it, but I'm not sure.  You're saying that in
```coq
Notation "x '.+1'" := (S x) (at level 10, format "x '.+1'") : mynat_scope.
Notation "f ( x )" := (f x) (at level 10, format "f ( x )") : app_scope.
```
the notation `.+1` takes priority over `f(x)`, even though the latter was declared more recently and neither scope is open?

Also, how is this a special case of the previous item?

Reply:
> This needs a more complete example.

I elaborated. Tell me if ok for you.

I otherwise took into account all other local corrections you proposed.